### PR TITLE
Add a way to disable PiP

### PR DESCRIPTION
**Pull request by beaufortfrancois**
This PR is about adding a hint to disable Picture In Picture for a specific video element as some may not want this for some videos.

WDYT?

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -155,6 +155,33 @@ the user agent MUST run the following steps:
...
+If the {{disablePictureInPicture}} attribute is present on the video element,
+the user agent SHOULD NOT play the video element in picture in picture or
+present any UI to do so.
+
+When the {{disablePictureInPicture}} attribute is added to a |video| element,
+the user agent SHOULD run these steps:
+
+1. Reject any pending promises returned by the {{requestPictureInPicture()}}
+    method with {{InvalidStateError}}.
+2. If |video| is playing in Picture in Picture:
```
Comment (mounirlamouri) [inline index.bs]: We should be able to use more normative language here. We do expose the current PIP element, right?

**Comment by jernoble** [comment]
@mounirlamouri 
> My main concern is that blocking the API with an attribute that the website would set is a bit weird.

Would it be less weird to you if the attribute reflected to a property? (Or are we not doing attribute reflection any longer?)

**Comment by mounirlamouri** [comment]
I think @beaufortfrancois's proposal had a property reflecting the attribute.

In general, I'm not strongly against this and if you would ship an attribute that would fully bock the API and the UA features (UI buttons, auto-pip, etc.), I would be fine having it in Blink and therefore in the spec.

File: index.bs [hunk 2]
```diff
@@ -155,6 +155,33 @@ the user agent MUST run the following steps:
...
+MUST reflect the content attribute of the same name.
+
+If the {{disablePictureInPicture}} attribute is present on the video element,
+the user agent SHOULD NOT play the video element in picture in picture or
+present any UI to do so.
+
+When the {{disablePictureInPicture}} attribute is added to a |video| element,
+the user agent SHOULD run these steps:
+
+1. Reject any pending promises returned by the {{requestPictureInPicture()}}
```
Comment (mounirlamouri) [inline index.bs]: Shouldn't you add steps in the enter picture in picture algorithm to reject if the attribute is set too?

Comment (beaufortfrancois) [reply index.bs]: Fixed in https://github.com/WICG/picture-in-picture/pull/28/commits/04ce8d3084193ab3c7ff7d23808eee0c59450cdb

File: index.bs [hunk 3]
```diff
@@ -125,11 +125,13 @@ invoked, the user agent MUST run the following steps:
     steps.
 2. If the algorithm is not <a>triggered by user activation</a>, throw a
     {{NotAllowedError}} and abort these steps.
-3. Let |pipWindow| be the Picture In Picture window that is <dfn>always on
+3. OPTIONALLY, if the {{disablePictureInPicture}} attribute is present on
+    |video|, throw a {{NotAllowedError}} and abort these steps.
```
Comment (mounirlamouri) [inline index.bs]: Shouldn't we stay consistent and use `InvalidStateError`?

**Pull request merged**